### PR TITLE
Dataset detail

### PR DESCRIPTION
**Pull request by nguyenlinhlinh**
## Description
This PR solves issue #762 and made some small changes to the ErrorPageComponent.
## Motivation
See issue #762 
## Fixes:
- View a private dataset without logging in.
   Screenshot 2021-05-05 at 09 43 05
- View public dataset while being logged in => return dataset details (was empty before)
- View private dataset that the user doesn't have access to.
Screenshot 2021-05-05 at 09 47 10
- Invalid url redirects to /404
Screenshot 2021-05-05 at 09 49 00
## Tests included/Docs Updated?

- [ ] Included for each change/fix?
- [x] Passing? (Merge will not be approved unless this is checked) 
- [ ] Docs updated?
- [ ] New packages used/requires npm install? 
- [ ] Toggle added for new features?
- [x] Requires update of catamel API?

### CODE REVIEW

File: src/app/app-routing/app-routing.module.ts [hunk 1]
```diff
@@ -257,8 +257,18 @@ export const routes: Routes = [
         component: LoginLayoutComponent,
         canActivate: [AuthGuard],
       },
+      {
+        path: "404",
+        component: ErrorPageComponent,
+        data: { errorTitle: "404 Page not found" , message: "Sorry, the page you are trying to view doesn't exists"}
```
Comment (henrikjohansson712) [inline src/app/app-routing/app-routing.module.ts:263]: For consistency, either replace the "does not" with "doesn't" here or the other way around in the other places.

File: src/app/datasets/anonymous-details-dashboard/anonymous-details-dashboard.component.html [hunk 2]
```diff
@@ -55,3 +55,11 @@ <h3>No datafiles linked to this dataset</h3>
     </mat-tab>
   </mat-tab-group>
 </ng-container>
+<ng-template #notfound>
+  <error-page
+    *ngIf="(loading$ | async) === false"
+    [errorTitle]="'Dataset not found'"
+    [message]="'The dataset you are trying to view is either does not exist or requires authentication.'"
```
Comment (henrikjohansson712) [inline src/app/datasets/anonymous-details-dashboard/anonymous-details-dashboard.component.html:62]: The "is" should be removed.

File: src/app/datasets/dataset-details-dashboard/dataset-details-dashboard.component.html [hunk 3]
```diff
@@ -131,3 +139,11 @@ <h3>Datablocks: {{ datablocks.length }}</h3>
     </mat-tab>
   </mat-tab-group>
 </ng-template>
+<ng-template [ngIf]="!dataset">
+  <error-page
+    *ngIf="(loading$ | async) === false"
+    [errorTitle]="'Dataset not found'"
+    [message]="'The dataset you are trying to view is either does not exist or you do not have permission to view it.'"
```
Comment (henrikjohansson712) [inline src/app/datasets/dataset-details-dashboard/dataset-details-dashboard.component.html:146]: The "is" should be removed.

**Review by henrikjohansson712 — CHANGES_REQUESTED** [review]
I like the look of the error pages, just a few remarks on the displayed text.

**Comment by nguyenlinhlinh** [comment]
Thank you. I will make changes.

**Review by henrikjohansson712 — APPROVED** [review]

**Pull request merged**